walkthroughs/ec2/ec2-cluster.yaml

```diff
@@ -0,0 +1,177 @@
+    Default: "subfuzion/aws-appmesh-proxy-route-manager:latest"
+
+  EnvoyImage:
+    Type: String
+    Description: Envoy image to use
+
+  AppMeshXdsEndpoint:
+    Type: String
+    Description: App Mesh XDS Endpoint Override
+    Default: ""
+
+  EnvoyLogLevel:
+    Type: String
+    Default: debug
+
+  ColorTellerImage:
+    Type: String
+    Description: Color Teller app image
+    
+  AppMeshMeshName:
+    Type: String
+    Description: Name of mesh
+
+  AppMeshEgressIgnoredIpCsv:
+    Type: String
+    Default: 169.254.170.2,169.254.169.254
+
+  ECSServicesDomain:
+    Type: String
+    Description: DNS namespace used by services e.g. default.svc.cluster.local
+
+Resources:
+
+  EC2SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Allow connections on specified ports
+      VpcId:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:VPC"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 9080
+          ToPort: 9080
+          CidrIp: 0.0.0.0/0
+
+  # EC2 instance for running Docker containers
+  EC2Instance:
+    Type: AWS::EC2::Instance
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
+    Properties:
+      ImageId: !Ref EC2AMI
+      InstanceType: !Ref InstanceType
+      SecurityGroupIds:
+        - !Ref EC2SecurityGroup
+      SubnetId:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:PublicSubnet1"
+      KeyName: !Ref KeyName
+      Tags:
+        - Key: Name
+          Value: !Sub ${EnvironmentName}-ec2-instance
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash -x
+
+            # CloudFormation support - use to signal cloudformation when finished
+            yum install -y aws-cfn-bootstrap
+
+            # golang support - use to build colorteller app
+            amazon-linux-extras install -y golang1.11
+            mkdir -p ~/go ~/go/bin ~/go/pkg ~/go/src/github.com/aws
+            GOPATH=~/go go get github.com/aws/aws-app-mesh-examples/examples/apps/colorapp/src/colorteller/
+            sudo mv ~/go/bin/colorteller /usr/local/bin
+            # TODO set up systemd service instead
+            SERVER_PORT=9080 COLOR=yellow /usr/local/bin/colorteller &
+
+            # Docker support - add ec2-user to docker group
+            amazon-linux-extras install -y docker
+            systemctl start docker
+            usermod -a -G docker ec2-user
+
+            # pull/cache images before any other updates to network
+            docker pull ${EnvoyImage}
+            docker pull ${ProxyRouterManagerImage}
+
+            # run Envoy proxy container
+            docker run
+              --name envoy
+              --detach
+              --net=host \
+              --user "1337" \
+              -e "APPMESH_VIRTUAL_NODE_NAME=mesh/${AppMeshMeshName}/virtualNode/colorteller-yellow-vn" \
```

Comment:
We should still templatize the node name

Reply:
Noted. It's consistent with [how we create envoy containers](https://github.com/aws/aws-app-mesh-examples/blob/master/examples/apps/colorapp/ecs/create-task-defs.sh#L50) in other samples,  but I'll take that task on as part of the refactoring project we're working on.